Evaluation/configs/derive_wp.yaml

```diff
@@ -3,31 +3,53 @@ wp_maker:
 
   wp_definitions:
     e:
-      wp_names: ["VVTight", "VTight", "Tight", "Medium", "Loose", "VLoose", "VVLoose", "VVVLoose"]
-      wp_eff: [0.6, 0.7, 0.8, 0.9, 0.95, 0.98, 0.99, 0.995]
+      VVTight: 0.6
+      VTight: 0.7
+      Tight: 0.8
+      Medium: 0.9 
+      Loose: 0.95
+      VLoose: 0.98
+      VVLoose: 0.99
+      VVVLoose: 0.995
     mu:
-      wp_names: ["Tight", "Medium", "Loose", "VLoose"]
-      wp_eff: [0.995, 0.998, 0.999, 0.9995]
+      Tight: 0.995
+      Medium: 0.998
+      Loose: 0.999
+      VLoose: 0.9995
     jet:
-      wp_names: ["VVTight", "VTight", "Tight", "Medium", "Loose", "VLoose", "VVLoose", "VVVLoose"]
-      wp_eff: [0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.95, 0.98]
+      VVTight: 0.4
+      VTight: 0.5 
+      Tight: 0.6
+      Medium: 0.7
+      Loose: 0.8
+      VLoose: 0.9
+      VVLoose: 0.95
+      VVVLoose: 0.98
     
   tpr_step: 0.0001 # will evenly sample grid of values from 0 to 1 with step=tpr_step
-  require_wp_vs_others: True # if in computation of WP thresholds for a given `vs_type` taus should pass the loosest WPs against remaining `vs_types` 
+  require_wp_vs_others: True # if in computation of WP thresholds for a given `vs_type` taus should pass WPs from `WPs_to_require` against remaining `vs_types` 
+  WPs_to_require:
+    e: VVVLoose
+    mu: VLoose
+    jet: VVVLoose
+  
+  # convergence params
+  epsilon: 1e-7
+  n_iterations: 100
+  verbose: False
 
 create_df:
   _target_: derive_wp.create_df
 
   # path to prediction files
-  _path_to_mlflow: ???
-  _experiment_id: ???
-  _run_id: ???
-  path_to_preds: "${create_df._path_to_mlflow}/${create_df._experiment_id}/${create_df._run_id}/artifacts/predictions/"
+  path_to_mlflow: ???
+  experiment_id: ???
+  run_id: ???
+  path_to_preds: "${create_df.path_to_mlflow}/${create_df.experiment_id}/${create_df.run_id}/artifacts/predictions/"
   pred_samples: # samples in `path_to_preds` to take taus from 
     GluGluHToTauTau_M125: 'eventTuple_1-7_pred.h5' # single files, list of files and "*" are supported 
   
   # paths to input ROOT files will be retrieved from corresponding pred_input_filemap.json
   input_branches: [ 'tau_pt', 'tau_eta', 'tau_decayMode' ]
   input_tree_name: taus
-  selection: '(tau_pt>=20) and (tau_pt<100)'
-  tau_types: ['e', 'mu', 'tau', 'jet']
\ No newline at end of file
+  selection: '(tau_pt>=20) and (tau_pt<100) and not (tau_decayMode in [5,6])'
```

Comment:
sorry, I forgot, aslo `abs(tau_dz) < 0.2` should be applied

Reply:
actually, what about `abs(tau_eta) < 2.3`, I think it also should be applied at this point?

Reply:
Yes, indeed :)